content/en/docs/reference/setup-tools/kubeadm/implementation-details.md

```diff
@@ -76,6 +76,20 @@ in a majority of cases, and the most intuitive location; other constant paths an
   - `front-proxy-ca.crt`, `front-proxy-ca.key` for the front proxy certificate authority
   - `front-proxy-client.crt`, `front-proxy-client.key` for the front proxy client
 
+## The kubeadm configuration API
+
+Most kubeadm commands support a `--config` flag which allows passing a configuration file from
+disk. The configuration file format follows the common Kubernetes API `apiVersion` / `kind` scheme,
+but is considered a component configuration API, similarly to that of the kubelet.
+
+Different kubeadm commands require a different `kind`. For example, `InitConfiguration` for `kubeadm init`, `JoinConfiguration` for `kubeadm join`, `UpgradeConfiguration` for `kubeadm upgrade` and `ResetConfiguration`
+for `kubeadm reset`.
+
+The command `kubeadm config migrate` can be used to migrate an older, deprecated API file
+to a newer API file.
```

Comment:
Is this right?
```suggestion
The command `kubeadm config migrate` can be used to migrate an older format configuration
file to a newer (current) configuration format. The kubeadm tool only supports migrating from
deprecated configuration formats to the current format.
```
I reworded into the active voice per the [style guide](https://kubernetes.io/docs/contribute/style/style-guide/).